app/models/manageiq/providers/redhat/infra_manager/provision/configuration.rb

```diff
@@ -27,7 +36,13 @@ def configure_container
       configure_cpu(rhevm_vm)
       configure_host_affinity(rhevm_vm)
       configure_network_adapters
-      configure_cloud_init
+
+      sysprep_option = get_option(:sysprep_enabled)
+      if sysprep_option == 'file'
+        configure_sysprep
+      elsif sysprep_option == 'fields'
```

Comment:
So we will only configure cloud-init if sysprep_enabled == fields? Sounds strange to me. What would happen if the user tries to use cloud-init without setting any sysprep-related option?

Reply:
Now we have a choice in the UI. We can go without a configuration or we need to use choose from the UI `specification` option and we can see other fields on the UI. It is not possible to use it without it.
I did not choose a name and I agree it is not that meaningful but this is how it works for vmware.

Reply:
So, if I understand correctly, the user will have an additional step in the UI in order to use cloud-init: to choose the "fields" option. After that the UI for cloud-init will be like it used to be? Or will it also require the user to upload a cloud-init configuration file? Can you maybe add some screenshots showing how it looks like now? Both for sysprep and cloud-init?

Reply:
Note that I don't object to that additional step, just want to make sure that the rest of the UI is like it used to be for cloud-init.

Reply:
UI structure and naming is exactly the same.